Chore/update branching doc (#19116)
supabase/supabase: apps/docs/pages/guides/platform/branching.mdx

  <figcaption>Each branch has a separate Supabase instance.</figcaption>
</figure>

> Supabase branching works like Git branching, with one main Production branch and others for proposing and merging changes into it.

Reviewer: ```suggestion
Supabase branching works like Git branching, with one main Production branch. Preview Branches branch off your Production branch and contain proposed changes. These changes can then be merged into Production. 
```

> When you create a Preview Branch (branched off your Production branch), the SQL migration files from your Production branch are run on the new Preview Branch. The Preview Branch then reflects your current Production branch schema.

Reviewer: ```suggestion
When you create a Preview Branch, the SQL migration files from your Production branch are run on the new Preview Branch. The Preview Branch then reflects your current Production branch schema.
```

Only *schema* changes are captured in preview branches.

Preview Branch instances contain no data by default. You must include a seed file to seed your preview instance with sample data when the Preview Branch is created. Future versions of Branching may allow for automated data seeding and cloning.

</Admonition>

## Git providers

To facilitate Supabase Branching, your Supabase project will be connected to a Git repository via an integration.

Supabase Branching works with [GitHub](#branching-with-github). Future iterations may support other Git providers. If you are interested in other Git providers or a non-Git-based workflow, join the discussions for [GitLab Branching, BitBucket Branching, and non-Git based Branching](https://github.com/orgs/supabase/discussions/18936).

### Branching with GitHub

Supabase Branching uses the Supabase GitHub integration to read files from your GitHub repository. With this integration, Supabase watches all commits, branches, and pull requests of your GitHub repository.

In Git, you have a Production Branch (typically this is `main`, `master`, `prod`, etc). This should also be your Supabase project's Production Branch.

> You can create a corresponding Preview Branch for any Gith Branch in your repository. Each time a new Preview Branch is created, the migrations in the Git branch of that Preview Branch are run on the Preview Branch. Just like in Git, a Preview Branch is an exact copy of the Production Branch at the time of creation.

Reviewer 2: I don't capitalize branch in 'Git branch' - Only 'Preview Branch' as it's one of our features.

Reviewer 2: ```suggestion
You can create a corresponding Preview Branch for any Git branch in your repository. Each time a new Preview Branch is created, the migrations in the Git branch of that Preview Branch are run on the Preview Branch. Just like in Git, a Preview Branch is an exact copy of the Production Branch at the time of creation.
```

> Just like in Git, a Preview Branch is an exact copy of the Production Branch at the time of creation.

Not sure this is true for everyone. It's generally the working practice though for many though.

#### Set up an existing project for branching

Before enabling branching on an existing project, make sure your database-related files are up to date. You need a current snapshot of your existing database, so newer migration files have a foundation to run on top of.

> <Admonition type="caution" label="Get setup before turning on branching">

Reviewer: ```suggestion
<Admonition type="caution" label="Get set up before turning on branching">
```

If your migration files are up to date with your production database, you don’t need to do anything. You can now turn on branching.

> ##### If you don't have migration files in GitHub:

Reviewer 2: huh? weird
